index.bs

```diff
@@ -7932,10 +7879,56 @@ steps given |request| and |response|:
 This event is emitted before a request is sent (either over the network or
 before it's handled by a serviceworker or a local cache).
 
+<div algorithm>
+The steps to check if <dfn>request originates in user context</dfn> given |request|
+and |user context| are:
+
+1. Let |settings| be |request|'s [=request/client=].
+
+1. Let |related navigables| be [=get related navigables=] with |settings|.
+
+1. For |navigable| in |related navigables|:
+
+  1. If |navigable|'s [=associated user context=] is |user context| return true.
+
+1. Return false.
+
+</div>
+
 <div algorithm>
 The [=remote end event trigger=] is the <dfn export>WebDriver BiDi before
 request sent</dfn> steps given |request|:
 
+1. For each |user context| in the [=set of user contexts=]:
+
+   1. If  [=request originates in user context=] steps with |request| and
+      |user context| return true:
+
+      1. For each |session| in [=active BiDI sessions=]
+
+         Note: |user context| can be in not more then one
+         [=user context to accept insecure certificates override map=].
+
+         1. If |session|'s [=user context to accept insecure certificates override map=]
+            [=map/contains|contain=] |user context|:
+
+            1. Let |accept insecure certificates| be |session|'s
+               [=user context to accept insecure certificates override map=][|user context|].
+
+            1. If |accept insecure certificates| is true:
+
+               1. Assert [=endpoint node=] supports accepting insecure TLS
+                  connections.
+
+               1. When running the [=Basic Certificate Processing=] steps for
+                  |request|, skip step a, along with any other
+                  implementation-defined certificate validation steps.
+
+            1. Otherwise, when running the [=Basic Certificate Processing=] steps
+               for |request|, perform step a, along with any other
+               implementation-defined certificate validation steps, must not be
+               skipped.
```

Comment:
```suggestion
            1. Otherwise, when running the [=Basic Certificate Processing=] steps
               for |request|, perform all steps along with any implementation-defined
               certificate validation steps.
```